tests/components/number/test_init.py

```diff
@@ -484,6 +489,7 @@ async def test_set_value(
 )
 async def test_temperature_conversion(
```

Comment:
Parametrization of the device class has ben added to this test, but all tests set the device class to temperature. Either revert the parametrization or add test cases making use of the device class parameter.